pallets/parachain-staking/src/lib.rs

```diff
@@ -1528,11 +1528,17 @@ pub mod pallet {
 					<DelayedPayouts<T>>::remove(paid_for_round);
 					<Points<T>>::remove(paid_for_round);
 
-					// remove up to 1000 candidates that did not produce any blocks for
-					// the given round
-					let _ = <AtStake<T>>::clear_prefix(paid_for_round, 1000, None);
+					// remove all candidates that did not produce any blocks for
+					// the given round. The weight is added based on the number of backend
+					// items removed.
+					let remove_result =
+						<AtStake<T>>::clear_prefix(paid_for_round, u32::max_value(), None);
```

Comment:
`u32::max_value()` shouldn't ever be relevant unless we have some other bug, in which case a lower value would prevent us from potentially going overweight.

This scenario is pretty vague, so I don't have a strong opinion about it, but I thought it was worth mentioning.